bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -664,28 +607,91 @@ private ContentPolicy getContentPolicy(@NotNull ResourceResolver resourceResolve
                     }
                 }
             }
-            ContentPolicy contentPolicy = policyManager.getPolicy(imageResource, request);
-            return contentPolicy;
+            return policyManager.getPolicy(imageResource, request);
         } else {
             LOGGER.warn("Could not get policy manager from resource resolver!");
         }
         return null;
     }
 
+    /**
+     * Create a image transformation map from the selector string of the given {@link SlingHttpServletRequest}. A valid selector can be
+     * - handler or
+     * - handler.width or
+     * - handler.quality.width
+     *
+     * @param request the current {@link SlingHttpServletRequest}
+     * @return Map with quality and width transformation parameter
+     */
+    private Map<String, Integer> getTransformationMap(SlingHttpServletRequest request, Resource component) throws IllegalArgumentException {
```

Comment:
What about just passing the selector string (instead of the request) and come up with some unit tests for this method specifically. You just need to mock the resource accordingly to either trigger the exception or not for invalid width.